### PR TITLE
Use c++11 strings instead of old c-style manual string manipulation.

### PR DESCRIPTION
**Pull request by baldersheim**
@havardpe PR

### CODE REVIEW

File: searchsummary/src/vespa/searchsummary/docsummary/keywordextractor.cpp [hunk 1]
```diff
@@ -42,32 +41,25 @@ KeywordExtractor::IsLegalIndexName(const char *idxName) const
...
-{
-    free(_prefix);
-}
+KeywordExtractor::IndexPrefix::~IndexPrefix() = default;
 
 bool
 KeywordExtractor::IndexPrefix::Match(const char *idxName) const
 {
-    return (strncmp(idxName, _prefix, _prefixLen) == 0);
+    return (strncmp(idxName, _prefix.c_str(), _prefix.size()) == 0);
```
Comment (havardpe) [inline searchsummary/src/vespa/searchsummary/docsummary/keywordextractor.cpp:56]: should use a string compare function here

**Review by havardpe — APPROVED** [review]
some minor comments

File: fastos/src/vespa/fastos/unix_file.cpp [hunk 2]
```diff
@@ -386,10 +386,10 @@ FastOS_UNIX_File::Delete(const char *name)
 bool
 FastOS_UNIX_File::Delete(void)
 {
-    assert(!IsOpened());
-    assert(_filename != nullptr);
+    assert( ! IsOpened());
+    assert( ! _filename.empty());
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply fastos/src/vespa/fastos/unix_file.cpp]: Good point, done.

File: searchsummary/src/vespa/searchsummary/docsummary/keywordextractor.cpp (hunk 1 above)
Comment (baldersheim) [reply searchsummary/src/vespa/searchsummary/docsummary/keywordextractor.cpp]: Fixed.

File: searchlib/src/vespa/searchlib/util/dirtraverse.cpp [hunk 3]
```diff
@@ -15,17 +15,16 @@ static int cmpname(const void *av, const void *bv)
         *(const DirectoryTraverse::Name *const *) av;
     const DirectoryTraverse::Name *const b =
         *(const DirectoryTraverse::Name *const *) bv;
-    return strcmp(a->_name, b->_name);
+    return strcmp(a->_name.c_str(), b->_name.c_str());
```
[the remark this answers is not in the document]
Comment (baldersheim) [reply searchlib/src/vespa/searchlib/util/dirtraverse.cpp]: Fixed

File: searchsummary/src/vespa/searchsummary/docsummary/keywordextractor.cpp [hunk 4]
```diff
@@ -42,32 +41,25 @@ KeywordExtractor::IsLegalIndexName(const char *idxName) const
...
-{
-    free(_prefix);
-}
+KeywordExtractor::IndexPrefix::~IndexPrefix() = default;
 
 bool
 KeywordExtractor::IndexPrefix::Match(const char *idxName) const
 {
-    return (strncmp(idxName, _prefix, _prefixLen) == 0);
+    return _prefix.compare(idxName) == 0;
```
Comment (toregge) [inline searchsummary/src/vespa/searchsummary/docsummary/keywordextractor.cpp:56]: This seems to be wrong.

Comment (baldersheim) [reply searchsummary/src/vespa/searchsummary/docsummary/keywordextractor.cpp:56]: Yes, it is. Fix on the way.

Comment (baldersheim) [reply searchsummary/src/vespa/searchsummary/docsummary/keywordextractor.cpp:56]: Fixed in #20227

**Pull request merged**